src/Adapter/Cache/Clearer/SymfonyCacheClearer.php

```diff
@@ -55,6 +57,11 @@ public function clear()
             return;
         }
 
+        if ($this->shutdownRegistered) {
+            return;
+        }
+
+        $this->shutdownRegistered = true;
         register_shutdown_function(function () use ($kernel) {
             // The cache may have been removed by Tools::clearSf2Cache, it happens during install
```

Comment:
I just realised it thinking to this class implementation but why do we use an anonymous function here ?
To forbid any external access ?
Is it a technical debt ? A recommended implementation in Prestashop ?

This is typically the kind of domain behavior I would like to test so extracting it should be interesting ^^'

Reply:
`SymfonyCacheClearer ` is not supposed to be instanciated manually, it is only used in the core via its service definition so yes I assume it will not be instanciated by a module Even if it did we can't anticipate any unexpected thing a module is able to do, the list is just limitless

Regarding the anonymous function it's just that `register_shutdown_function` requires a callback method, so using an anonymous function is as good as a private function anyway so it's not worth blocking this and it's not even the purpose of the PR or the modification of this class